engine/src/main/java/org/terasology/world/block/BlockBuilderHelper.java

```diff
@@ -21,23 +21,101 @@
 import org.terasology.world.block.loader.SectionDefinitionData;
 import org.terasology.world.block.shapes.BlockShape;
 
+/**
+ * A helper class for building and registering blocks with specific properties.
+ * Intended for use with Block Family related code.
+ * <p>
+ * The core features of this helper is to convert from definition data, into a Block
+ * and to register this block into the engine for use.
+ */
 public interface BlockBuilderHelper {
 
+    /**
+     * {@inheritDoc}
+     *
+     * @see #constructSimpleBlock(BlockFamilyDefinition, BlockShape, String, BlockUri, BlockFamily)
+     */
     Block constructSimpleBlock(BlockFamilyDefinition definition, BlockUri uri, BlockFamily blockFamily);
 
+    /**
+     * {@inheritDoc}
+     *
+     * @see #constructSimpleBlock(BlockFamilyDefinition, BlockShape, String, BlockUri, BlockFamily)
+     */
     Block constructSimpleBlock(BlockFamilyDefinition definition, BlockShape shape, BlockUri uri, BlockFamily blockFamily);
 
+    /**
+     * {@inheritDoc}
+     *
+     * @see #constructSimpleBlock(BlockFamilyDefinition, BlockShape, String, BlockUri, BlockFamily)
+     */
     Block constructSimpleBlock(BlockFamilyDefinition definition, String section, BlockUri uri, BlockFamily blockFamily);
 
+    /**
+     * Constructs a basic block from the data provided. This block is assumed to have no rotations, and to use reasonable defaults
```

Comment:
What are "reasonable defaults"? What kind of defaults are we talking about/defaults for what?

Reply:
I don't know enough about block creation to answer this adequately. All kinds of things are setup using the parameters. I assume there are block properties independent of the specified parameters that would be initialized "using reasonable defaults". It's an interface method, therefore it has an abstract description. The claim is to produce a useful block without having to specify all the properties its gonna have.